@neithernut Can you tell what this error means (from the build):

```
/home/travis/build/matthiasbeyer/libreset/src/libreset/ht.c: In function ‘ht_find’:
/home/travis/build/matthiasbeyer/libreset/src/libreset/ht.c:41:5: error: left shift count >= width of type [-Werror]
     return &ht->buckets[hash / ((CONSTPOW_TWO(BITCOUNT(hash)) / ht->nbuckets))];
```

and how to fix it?

---

By the way: You wrote in the paper that we use the most significant **bits**, but you did not state how many bits...

--- a/src/libreset/ht.c
+++ b/src/libreset/ht.c
@@ -21,8 +21,9 @@ ht_destroy(
     struct ht* ht
 ) {
     if (ht) {
-        for (; ht->nbuckets > 0; ht->nbuckets--) {
-            avl_destroy(&ht->buckets[ht->nbuckets - 1].avl);
+        size_t i = ht_nbuckets(ht);
+        while (i--) {
+            avl_destroy(&ht->buckets[i].avl);
         }
         free(ht->buckets);
         free(ht);

--- a/src/libreset/ht.c
+++ b/src/libreset/ht.c
@@ -9,8 +9,8 @@ ht_init(
     size_t n
 ) {
     if (ht) {
-        ht->buckets = calloc(n, sizeof(*ht->buckets));
-        ht->nbuckets = n;
+        ht->buckets = calloc(CONSTPOW_TWO(n), sizeof(*ht->buckets));
+        ht->sizeexp = n;
     }
 
     return ht;

--- a/src/libreset/ht.h
+++ b/src/libreset/ht.h
@@ -45,9 +45,11 @@ struct ht_bucket {
 
 /**
  * Hashtable type
+ *
+ * The type for the hashtable, holding buckets and size.
  */
 struct ht {
-    struct ht_bucket* buckets;
+    struct ht_bucket* buckets; //!< The buckets of the hashtable
     size_t nbuckets;
 };
 

--- a/src/libreset/ht.h
+++ b/src/libreset/ht.h
@@ -64,7 +64,7 @@ struct ht {
 struct ht*
 ht_init(
     struct ht* ht, //!< The hashtable object to initialize
-    size_t n //!< The initial number of buckets
+    size_t n //!< Power, 2 must be raised to, to calc the number of buckets.
 );
 
 /**

